docs/architecture/adr-018-network-upgrades.md

```diff
@@ -29,40 +30,48 @@ One of the main difficulties of social upgrades when using tendermint consensus
 
-### Phase 2: Signaled Upgrade Height
+For better performance, `VoteExtensions` should be modified such that empty messages don't require a signature (which is currently the case for v0.38 of [CometBFT](https://github.com/cometbft/cometbft/blob/91ffbf9e45afb49d34a4af91b031e14653ee5bd8/privval/file.go#L324))
 
-Preconfigured upgrade paths are vulnerable to halts. There is no indication that a quorum has in fact upgraded and that when the proposer proposes the block with the latest version, that consensus will be reached. To mitigate this risk, the upgrade height can instead be signaled by validators. Vote Extensions may appear as a good tool for this but it is inefficient to continually signal every height. Validators should only need to signal once. There are two possible approaches:
+#### Alternatives
+
+There are two alternative approaches that were considered:
 
 - **Off-chain**: A new p2p reactor is introduced whereby validators sign a message indicating they are now running a new binary and are ready to switch. Once a proposer has received a quorum plus some predefined grace period, they will propose a block with the new version and the rest of the network will vote accordingly. This approach means that the application doesn't have control but rather has to listen for changes in the app version. This also requires a change to the `PrivValidator` interface to be able to sign the new message.
 - **On-chain**: Upon upgrading to a new binary, the node will submit a transaction signalling it's ability to switch version. Again after a quorum is reached and some grace period, the `upgrade` module would trigger the app version change in `EndBlock`. The drawback with this approach is that this would probably require gas to submit in order to avoid spamming the network and wouldn't necessarily be automatic i.e. nodes could upgrade and forget to signal.
 
+### Future Work: Downgrading
+
+A more resilient system will have the option for a coordinated downgrade. This doesn't necessarily need to be because of a liveness or safety bug but could simply arise because of a degradation in service. Coordination can either be done through the configured upgrade heights or through the aforementioned signalling mechanism. The downgrade process is similar to the upgrade process. The proposer will propose a block with the `MsgVersionChange` to downgrade to the previous version. The validators will vote on this so long as it matches their local configured upgrade schedule and if consensus is reached, the app version will be downgraded at `EndBlock`. The next proposer will propose a block with the new version and the network will continue on as normal.
```

Comment:
> be done through the configured upgrade heights

What does this look like with the upgrade ranges described earlier? Does the upgrade module try to upgrade to the highest version number if there are overlapping ranges?

Reply:
Say you are at v10. To downgrade you would simply add v9 -> [100, 200] to your config. Thus you would propose and accept a downgrade over that height range